tests/gadget-zero/usb-gadget0.c

```diff
@@ -254,6 +307,28 @@ static void gadget0_tx_cb_loopback(usbd_device *usbd_dev, uint8_t ep)
 	/* TODO - unimplemented - consult linux source on proper behaviour */
 }
 
+static void gadget0_rx_cb_interrupt(usbd_device *usbd_dev, uint8_t ep)
+{
+	static uint8_t buf[64] __attribute__ ((aligned(4)));
+	int len = usbd_ep_read_packet(usbd_dev, ep, buf, 64);
+
+	ER_DPRINTF("interrupt rx %x %d\n", ep, len);
+
+	if (len == 0)
+		return;
+	switch (buf[0]) {
+	case 0: /* delay */ {
+		if (len < 5)
+			return;
+		uint32_t delay = buf[1] | ((uint32_t) buf[2] << 8) | ((uint32_t) buf[3] << 16) | ((uint32_t) buf[4] << 24);
```

Comment:
given you have control over the byte order when you made the packet, why not make it so you can just cast this here?